### PR TITLE
refactor(eos_cli_config_gen): Port Channel

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
refactor port-channels.j2

## How to test

run molecule scenarios

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2 [hunk 1]
```diff
@@ -1,183 +1,174 @@
...
    switchport trunk allowed vlan {{ port_channel_interfaces[port_channel_interface].vlans }}
-{%         endif %}
-{%         if port_channel_interfaces[port_channel_interface].native_vlan is defined and port_channel_interfaces[port_channel_interface].mode == "trunk" %}
+{%     endif %}
+{%     if port_channel_interfaces[port_channel_interface].native_vlan is arista.avd.defined and port_channel_interfaces[port_channel_interface].mode is arista.avd.defined("trunk") %}
    switchport trunk native vlan {{ port_channel_interfaces[port_channel_interface].native_vlan }}
-{%         endif %}
-{%         if port_channel_interfaces[port_channel_interface].mode is defined and port_channel_interfaces[port_channel_interface].mode == "trunk" %}
+{%     endif %}
+{%     if port_channel_interfaces[port_channel_interface].mode is arista.avd.defined and port_channel_interfaces[port_channel_interface].mode is arista.avd.defined("trunk") %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2]: Can be reduced to only `port_channel_interfaces[port_channel_interface].mode is arista.avd.defined("trunk")`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/virtual-source-nat.j2 [hunk 2]
```diff
@@ -6,4 +6,4 @@
 ip address virtual source-nat vrf {{ vrf }} address {{ virtual_source_nat_vrfs[vrf].ip_address }}
 {%         endif %}
 {%     endfor %}
-{% endif %}
\ No newline at end of file
+{% endif %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/virtual-source-nat.j2:9]: Empty new line is missing

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/virtual-source-nat.j2:9]: there is one, you just can see it on Github review!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2 (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2]: fixed with commit c97751c

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/virtual-source-nat.j2 (hunk 2 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/virtual-source-nat.j2:9]: There is one you just can't see in GitHub review! I've been caught by this before as well... when there is no new line you see a (-) red sign on the last line. It was commit c9aee95

**Review by titom73 — APPROVED** [review]
LGTM!

**Pull request merged**